source/develop/release-management/features/sla/hosted-engine-new-deployment.html.md

```diff
@@ -0,0 +1,152 @@
+---
+title: hosted-engine-new-deployment
+category: feature
+authors: jtokar
+feature_name: Provide ansible script to install Hosted Engine
+feature_modules: sla
+feature_status: in development 
+---
+
+# Hosted Engine Deployment 
+
+## Motivation 
+
+Today we are using the ovirt-hosted-engine-setup project to deploy hosted engine. The project contains the code to add storage, 
+disks and the vm. All of this code already exists in the ovirt-engine project. By reusing that code we can reduce bugs and unify
+the different engine installations. 
+
+## The change 
+
+The new process will use the current engine code to create all entities neeeded for successfull hosted engine deployment.   
+A local bootstrap vm with the engine appliance will be created on the host. After running engine setup on that vm we will use the 
+engine api to add a host (same one we are running on), storage domain, storage disks and finally a vm that will later become the 
+hosted engine vm in the engine (thus eliminating the need for importing it).   
+Once all the entities are created we can shut down the bootstrap vm, copy its disk to the disk we created using the engine, create 
+the hosted engine configuration files, start the agent and the broker and the hosted engine vm will start.   
+
+## How 
+
+All of this will be implemented using an Ansible script.  
+The current answer file will need to be converted to an Ansible vars files.   
+The configuration files will be created using Ansible templates and then packaged and copied to the correct location. 
+
+To run the script user shoud install ansible and run:   
+    ansible-playbook hosted-engine.yml   
+The script will install all the packages and will start the deployment.   
+The vars file needs to be populated manually for now. 
+
+## Issues
+
+1. Host installation fails during network setup stage  
+   The reason for this is that vdsm setup removes the routing table.   
+   Workaround: rewrite the routing table after the host gets to the "non operational" state and then save the network configuration 
+   using vdsm-client. This causes the host to go to "up" state almost immediatly.   
+   Permanent solution: handle this in host deploy code.   
+2. Adding hosted engine disk takes a long time   
+   This might not be a problem if using a cow format and not raw. But this wasn't tested yet.   
+3. When using fqdn that resolves to a real ip the bootstrap vm tries to access the real ip and the installation fails 
+   Workaround: add a line to /etc/hosts on the host with the fqdn and the vm temporary ip.  
+               add a line to /etc/hosts on the vm with localhost and fqdn - not implemented.   
+   After successfull installation those lines need to be removed.   
+4. When hosted engine with iscsi domain the storage domain creation fails  
```

Comment:
"When the Hosted Engine is with an ISCSI domain, the storage domain creation fails."